src/controllers/midi/midisourceclock.cpp

```diff
@@ -0,0 +1,117 @@
+#include "controllers/midi/midimessage.h"
+#include "controllers/midi/midisourceclock.h"
+#include "util/math.h"
+
+bool MidiSourceClock::handleMessage(unsigned char status) {
+    switch (status) {
+    case MIDI_START:
+        start();
+        return true;
+    case MIDI_STOP:
+        stop();
+        return true;
+    case MIDI_TIMING_CLK:
+        tick();
+        return true;
+    default:
+        return false;
+    }
+}
+
+void MidiSourceClock::start() {
+    m_bRunning = true;
+    m_iFilled = 0;
+    m_iRingBufferPos = 0;
+}
+
+void MidiSourceClock::stop() {
+    m_bRunning = false;
+}
+
+void MidiSourceClock::tick() {
```

Comment:
That would be a different implementation, so it would be done in a separate class, and we could update the API at that time with minimal refactoring.  No need to engage in speculative coding.